pulp_ansible/app/tasks/collections.py

```diff
@@ -378,12 +385,50 @@ def _build_url(path_or_url):
                         if result.get("version") and not download_url:
                             version_url = _build_url(result["href"])
                             not_done.update([remote.get_downloader(url=version_url).run()])
+                            additional_metadata[version_url] = {
+                                "docs_blob_url": f"{version_url}docs-blob/"
+                            }
 
                         if download_url:
+                            metadata = additional_metadata.get(_build_url(data["href"]), {})
+                            data["docs_blob_url"] = metadata.get("docs_blob_url")
                             yield data
                             progress_bar.increment()
 
 
+class DocsBlobDownloader(ArtifactDownloader):
```

Comment:
First off, it's really great you've not duplicated the code of `ArtifactDownloader`. I can see now that pulpcore.plugin would do well to make this functionality more re-usable. I think the only issue I see is that the progress reporting from `ArtifactDownloader` is not customizable [here](https://github.com/pulp/pulpcore/blob/master/pulpcore/plugin/stages/artifact_stages.py#L139).

So I wrote this for us to solve https://pulp.plan.io/issues/7464

I don't think the "incorrect progress reporting" problem should stop us from merging this and releasing 0.3.0. We should fix https://pulp.plan.io/issues/7464 with pulpcore 3.7 and switch `DocsBlogDownloader` to use `GenericDownloader` with the correct error message once 3.7 is GA.